File: rb/features.rb
# EnergyChartsApi2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module EnergyChartsApi2Features
  def self.make_feature(name)
    case name
    when "base"
      EnergyChartsApi2BaseFeature.new
    when "test"
      EnergyChartsApi2TestFeature.new
    else
      EnergyChartsApi2BaseFeature.new
    end
  end
end

File: py/features.py
# EnergyChartsApi2 SDK feature factory

from feature.base_feature import EnergyChartsApi2BaseFeature
from feature.test_feature import EnergyChartsApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: EnergyChartsApi2BaseFeature(),
        "test": lambda: EnergyChartsApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
